include/vrv/scoringupfunctor.h

```diff
@@ -0,0 +1,102 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        scoringupfunctor.h
+// Author:      Martha E. M. Thomae Elias
+// Created:     2024
+// Copyright (c) Authors and others. All rights reserved.
+/////////////////////////////////////////////////////////////////////////////
+///
+#ifndef __VRV_SCORINGUPFUNCTOR_H__
+#define __VRV_SCORINGUPFUNCTOR_H__
+
+#include "functor.h"
+
+namespace vrv {
+
+//----------------------------------------------------------------------------
+// ScoringUpFunctor
+//----------------------------------------------------------------------------
+
+/**
+ * This class...
+ */
+class ScoringUpFunctor : public Functor {
+public:
+    /**
+     * @name Constructors, destructors
+     */
+    ///@{
+    ScoringUpFunctor();
+    virtual ~ScoringUpFunctor() = default;
+    ///@}
+
+    /*
+     * Abstract base implementation
+     */
+    bool ImplementsEndInterface() const override { return true; }
```

Comment:
You can return `false` here, if the functor does not override any of the `VisitElementEnd` methods. In this case the functor skips those methods during execution.

```suggestion
    bool ImplementsEndInterface() const override { return false; }
```